src/autoSoundConfig.ml

```diff
@@ -0,0 +1,86 @@
+
+(** Automatically turning on analyses required to ensure soundness
+    based on a given specification (e.g., SV-COMP specification)
+    or programming idioms (e.g., longjump) in the analyzed code,
+    but only when it is possible to do so automatically.
+    This does not fully exempt from the need for manual configuration.
+*)
+
+open GobConfig
+open AutoTune
+
+let logEnablingAnalyses spec ana = Logs.info "Specification: %s -> enabling soundness analyses \"%s\"" (Svcomp.Specification.to_string [spec]) (String.concat ", " ana)
+
+let enableOptions options = 
+  let enableOpt option = 
+    Logs.info "Setting \"%s\" to true" option;
+    set_bool option true
+  in
+  List.iter enableOpt options
+
+(** Selecting sound configurations based on SV-COMP specification.
+    Activating the analyses and enabling options that are
+    required for analyzing the property defined in the specification.
+    TODO: have only one function for matching all specifications and find a place where it can be called.
+*)
+let enableAnalysesForMemSafetySpecification (spec: Svcomp.Specification.t) =
+  match spec with
+  | ValidFree -> (* Enable the soundness analyses for ValidFree spec *)
+    let analyses = ["base"; "useAfterFree"] in
+    logEnablingAnalyses spec analyses;
+    enableAnalyses analyses
```

Comment:
We should have some helper function to do both the logging and the enabling to reduce code duplication. For example, to write it like this:
```suggestion
    enableSpecAnalyses spec ["base"; "useAfterFree"]
```